provider/infoblox/infoblox.go

```diff
@@ -412,17 +441,19 @@ func (p *ProviderConfig) ApplyChanges(ctx context.Context, changes *plan.Changes
 
 func (p *ProviderConfig) zones() ([]ibclient.ZoneAuth, error) {
 	var res, result []ibclient.ZoneAuth
-	obj := ibclient.NewZoneAuth(
-		ibclient.ZoneAuth{
-			View: p.view,
-		},
-	)
-	err := p.client.GetObject(obj, "", nil, &res)
+	obj := ibclient.NewZoneAuth(ibclient.ZoneAuth{})
+	qp := ibclient.NewQueryParams(
+		false, map[string]string{"view": p.view})
+	err := p.client.GetObject(obj, "", qp, &res)
 	if err != nil && !isNotFoundError(err) {
 		return nil, err
 	}
 
 	for _, zone := range res {
+		if p.fqdnRegEx != nil && !p.fqdnRegEx.MatchString(zone.Fqdn) {
+			continue
+		}
```

Comment:
FQDNRegExp is used in the RequestBuilder passed to the Infoblox Connector and here. So we filter twice using the same flag. I thought that the idea was to filter on the server side using the regexp flag and on the client side using domain and zone-id filters. If so then this code is not needed.